--- qtbase-opensource-src-5.1.0/mkspecs/features/qt_module.prf.old	2013-07-23 08:20:36.843930264 +0000
+++ qtbase-opensource-src-5.1.0/mkspecs/features/qt_module.prf	2013-07-23 08:22:52.285974034 +0000
@@ -140,18 +140,36 @@
       rplbase = $$dirname(_QMAKE_SUPER_CACHE_)/[^/][^/]*
    else: \
       rplbase = $$MODULE_BASE_OUTDIR
-   include_replace.match = $$rplbase/include
-   include_replace.replace = $$[QT_INSTALL_HEADERS/raw]
-   lib_replace.match = $$rplbase/lib
-   lib_replace.replace = $$QMAKE_PKGCONFIG_LIBDIR
-   QMAKE_PKGCONFIG_INSTALL_REPLACE += include_replace lib_replace
+   pkgconfig_include_replace.match = $$rplbase/include
+   pkgconfig_include_replace.replace = "\$$\\{includedir}"
+   pkgconfig_lib_replace.match = $$rplbase/lib
+   pkgconfig_lib_replace.replace = "\$$\\{libdir}"
+   !exists($$[QT_SYSROOT]) {
+      include_replace.match = $$rplbase/include
+      include_replace.replace = $$[QT_INSTALL_HEADERS/raw]
+      lib_replace.match = $$rplbase/lib
+      lib_replace.replace = $$[QT_INSTALL_LIBS/raw]
+      lafile_replace.match = $$rplbase
+      lafile_replace.replace = "$$[QT_INSTALL_PREFIX/raw]"
+   } else {
+      # include_replace and lib_replace are duplicate, but we don't want to
+      # make QMAKE_PKGCONFIG_INSTALL_REPLACE, QMAKE_PRL_INSTALL_REPLACE conditional
+      include_replace.match = $$rplbase
+      include_replace.replace = $$[QT_SYSROOT]$$[QT_INSTALL_PREFIX/raw]
+      lib_replace.match = $$rplbase
+      lib_replace.replace = $$[QT_SYSROOT]$$[QT_INSTALL_PREFIX/raw]
+      lafile_replace.match = $$rplbase
+      lafile_replace.replace = "=$$[QT_INSTALL_PREFIX/raw]"
+   }
+
+   QMAKE_PKGCONFIG_INSTALL_REPLACE += pkgconfig_include_replace pkgconfig_lib_replace
    QMAKE_PRL_INSTALL_REPLACE += include_replace lib_replace
 }
 
 unix {
    CONFIG += create_libtool explicitlib
    QMAKE_LIBTOOL_LIBDIR = $$QMAKE_PKGCONFIG_LIBDIR
-   QMAKE_LIBTOOL_INSTALL_REPLACE += include_replace lib_replace
+   QMAKE_LIBTOOL_INSTALL_REPLACE += lafile_replace
 }
 
 unix|win32-g++* {
