--- qtbase-opensource-src-5.1.1/mkspecs/win32-g++/qmake.conf~orig	2014-03-10 13:18:52.166561069 +0100
+++ qtbase-opensource-src-5.1.1/mkspecs/win32-g++/qmake.conf	2014-03-10 13:18:50.222561071 +0100
@@ -13,7 +13,7 @@
 QMAKE_PLATFORM          = win32 mingw
 CONFIG                 += debug_and_release debug_and_release_target precompile_header
 DEFINES                += UNICODE
-QMAKE_COMPILER_DEFINES += __GNUC__ WIN32
+QMAKE_COMPILER_DEFINES += __GNUC__ WIN32 Q_OS_WIN
 
 QMAKE_EXT_OBJ           = .o
 QMAKE_EXT_RES           = _res.o
@@ -25,7 +25,7 @@
 QMAKE_LEXFLAGS          =
 QMAKE_YACC              = byacc
 QMAKE_YACCFLAGS         = -d
-QMAKE_CFLAGS            = -pipe -fno-keep-inline-dllexport
+QMAKE_CFLAGS            = -pipe -fno-keep-inline-dllexport -DUNICODE -D_UNICODE
 QMAKE_CFLAGS_DEPS       = -M
 QMAKE_CFLAGS_WARN_ON    = -Wall -Wextra
 QMAKE_CFLAGS_WARN_OFF   = -w
