--- qtbase-opensource-src-5.1.1/configure.old	2014-03-12 09:22:08.287336576 +0100
+++ qtbase-opensource-src-5.1.1/configure	2014-03-12 09:23:56.919335001 +0100
@@ -291,16 +291,6 @@
     getSingleQMakeVariable "$1" "$specvals"
 }
 
-# OE qmake.conf is reading some variables from shell env
-# read them from qmake.conf, replace qmake () syntax with shell and eval
-getQEvalMakeConf()
-{
-    VAL=`getQMakeConf "$1" | sed -n 's/$[(]\([0-9a-zA-Z_]*\)[)]/$\1/pg'`
-    EVAL=`eval "echo ${VAL}"`
-#    echo "Running getQEvalMakeConf: var='$1', val='`getQMakeConf \"$1\"`, val-sed='$VAL', eval='$EVAL'" >&2
-    eval "echo ${VAL}"
-}
-
 getXQMakeConf()
 {
     if [ -z "$xspecvals" ]; then
@@ -310,16 +300,6 @@
     getSingleQMakeVariable "$1" "$xspecvals"
 }
 
-# OE qmake.conf is reading some variables from shell env
-# read them from qmake.conf, replace qmake () syntax with shell and eval
-getXQEvalMakeConf()
-{
-    VAL=`getXQMakeConf "$1" | sed -n 's/$[(]\([0-9a-zA-Z_]*\)[)]/$\1/pg'`
-    EVAL=`eval "echo ${VAL}"`
-#    echo "Running getXQEvalMakeConf: var='$1', val='`getXQMakeConf \"$1\"`, val-sed='$VAL', eval='$EVAL'" >&2
-    eval "echo ${VAL}"
-}
-
 compilerSupportsFlag()
 {
     cat >conftest.cpp <<EOF
@@ -836,18 +816,6 @@
 # initalize variables
 #-------------------------------------------------------------------------------
 
-# Export all OE variables for qmake.conf from shell env to QMakeVars
-OE_VARIABLES="AR CC CFLAGS COMPILER CXX CXXFLAGS LDFLAGS LINK QT_CONFIG STRIP"
-for varname in $OE_VARIABLES; do
-    qmakevarname="${varname}"
-    cmd=`echo \
-'if [ -n "\$OE_QMAKE_'${varname}'" ]; then
-    QMakeVar set OE_QMAKE_'${qmakevarname}' "\$OE_QMAKE_'${varname}'"
-#    echo "Exporting OE_QMAKE_'${qmakevarname}' value=\"\$OE_QMAKE_'${varname}'\"" >&2
-fi'`
-    eval "$cmd"
-done
-
 SYSTEM_VARIABLES="RANLIB STRIP OBJDUMP LD CC CXX CFLAGS CXXFLAGS LDFLAGS"
 
 # Use CC/CXX to run config.tests
@@ -2913,7 +2881,7 @@
     CFG_QGTKSTYLE=no
 fi
 
-QMAKE_CONF_COMPILER=`getXQEvalMakeConf QMAKE_CXX`
+QMAKE_CONF_COMPILER=`getXQMakeConf QMAKE_CXX`
 
 TEST_COMPILER=$QMAKE_CONF_COMPILER
 
@@ -2948,7 +2916,7 @@
         exit 1
     fi
 fi
-TEST_COMPILER_CXXFLAGS=`getXQEvalMakeConf QMAKE_CXXFLAGS`
+TEST_COMPILER_CXXFLAGS=`getXQMakeConf QMAKE_CXXFLAGS`
 
 GCC_MACHINE_DUMP=
 case "$TEST_COMPILER" in *g++) GCC_MACHINE_DUMP=$($TEST_COMPILER -dumpmachine);; esac
@@ -3978,13 +3946,6 @@
     getQMakeConf "$1" | echo ${2-$1} = `if [ -n "$3" ]; then sed "$3"; else cat; fi` >> "$mkfile"
 }
 
-# OE qmake.conf is reading some variables from shell env
-# read them from qmake.conf, replace qmake () syntax with shell and eval
-setBootstrapEvalVariable()
-{
-    getQEvalMakeConf "$1" | echo ${2-$1} = `if [ -n "$3" ]; then sed "$3"; else cat; fi` >> "$mkfile"
-}
-
 
 # build qmake
 if true; then ###[ '!' -f "$outpath/bin/qmake" ];
@@ -4024,11 +3985,11 @@
         fi
 
         [ "$CFG_SILENT" = "yes" ] && CC_TRANSFORM='s,^,\@,' || CC_TRANSFORM=
-        setBootstrapEvalVariable QMAKE_CC CC "$CC_TRANSFORM"
-        setBootstrapEvalVariable QMAKE_CXX CXX "$CC_TRANSFORM"
-        setBootstrapEvalVariable QMAKE_CFLAGS
-        setBootstrapEvalVariable QMAKE_CXXFLAGS
-        setBootstrapEvalVariable QMAKE_LFLAGS
+        setBootstrapVariable QMAKE_CC CC "$CC_TRANSFORM"
+        setBootstrapVariable QMAKE_CXX CXX "$CC_TRANSFORM"
+        setBootstrapVariable QMAKE_CFLAGS
+        setBootstrapVariable QMAKE_CXXFLAGS
+        setBootstrapVariable QMAKE_LFLAGS
 
         if [ "$CFG_RELEASE_QMAKE" = "yes" ]; then
             setBootstrapVariable QMAKE_CFLAGS_RELEASE
