--- qtbase-opensource-src-5.1.0/src/src.pro.old	2013-07-24 05:44:03.372441808 +0000
+++ qtbase-opensource-src-5.1.0/src/src.pro	2013-07-24 05:44:15.512626574 +0000
@@ -101,7 +101,7 @@
 src_android.subdir = $$PWD/android
 
 # this order is important
-SUBDIRS += src_tools_bootstrap src_tools_moc src_tools_rcc src_corelib
+SUBDIRS += src_tools_bootstrap src_tools_moc src_tools_rcc src_tools_uic src_corelib
 win32:SUBDIRS += src_winmain
 SUBDIRS += src_network src_sql src_xml src_testlib
 contains(QT_CONFIG, dbus) {
