--- qt-mobility-opensource-src-1.2.0-beta1/plugins/plugins.pro.orig	2011-04-13 10:26:47.136686351 +0200
+++ qt-mobility-opensource-src-1.2.0-beta1/plugins/plugins.pro	2011-04-13 10:26:54.768238346 +0200
@@ -11,4 +11,4 @@
 contains(mobility_modules,location): SUBDIRS += geoservices
 contains(mobility_modules,versit): SUBDIRS += versit
 
-contains(QT_CONFIG,declarative): SUBDIRS += declarative
+SUBDIRS += declarative

--- qt-mobility-opensource-src-1.2.0-beta1/plugins/multimedia/gstreamer/gstreamer.pro.orig	2011-04-13 10:27:15.342422399 +0200
+++ qt-mobility-opensource-src-1.2.0-beta1/plugins/multimedia/gstreamer/gstreamer.pro	2011-04-13 10:27:22.863951996 +0200
@@ -78,24 +78,6 @@
     qgstreamervideoinputdevicecontrol.cpp \
     gstvideoconnector.c
 
-
-!win32:!embedded:!mac:!symbian:!simulator:!contains(QT_CONFIG, qpa) {
-    LIBS += -lXv -lX11 -lXext
-
-    HEADERS += \
-        qgstreamervideooverlay.h \
-        qgstreamervideowindow.h \
-        qgstreamervideowidget.h \
-        qx11videosurface.h \
-        qgstxvimagebuffer.h
-
-    SOURCES += \
-        qgstreamervideooverlay.cpp \
-        qgstreamervideowindow.cpp \
-        qgstreamervideowidget.cpp \
-        qx11videosurface.cpp \
-        qgstxvimagebuffer.cpp
-}
 include(mediaplayer/mediaplayer.pri)
 include(mediacapture/mediacapture.pri)
 
