--- qt-mobility-opensource-src-1.1.0-tp/src/multimedia/qmediaplaylistnavigator.h.orig	2011-04-08 11:41:28.374654291 +0200
+++ qt-mobility-opensource-src-1.1.0-tp/src/multimedia/qmediaplaylistnavigator.h	2011-04-08 11:42:10.193217881 +0200
@@ -87,6 +87,7 @@
     void activated(const QMediaContent &content);
     void currentIndexChanged(int);
     void playbackModeChanged(QMediaPlaylist::PlaybackMode mode);
+    void currentItemChanged(QMediaContent content);
 
     void surroundingItemsChanged();

--- qt-mobility-opensource-src-1.1.0-tp/src/multimedia/qvideowidget.h.orig	2011-04-08 11:42:24.236092832 +0200
+++ qt-mobility-opensource-src-1.1.0-tp/src/multimedia/qvideowidget.h	2011-04-08 11:43:03.604150455 +0200
@@ -97,6 +97,7 @@
     void contrastChanged(int contrast);
     void hueChanged(int hue);
     void saturationChanged(int saturation);
+    void aspectRatioModeChanged(Qt::AspectRatioMode mode);
 
 protected:
     bool event(QEvent *event);
