Author: Morten Thunberg Svendsen <Morten.ThunbergSvendsen@prevas.dk>
Date:   Tue Apr 15 15:15:15 2015 +0200

Log X Visual ID debug info to syslog


Index: qtbase-opensource-src-5.3.1/src/platformsupport/eglconvenience/qxlibeglintegration.cpp
===================================================================
--- qtbase-opensource-src-5.3.1.orig/src/platformsupport/eglconvenience/qxlibeglintegration.cpp
+++ qtbase-opensource-src-5.3.1/src/platformsupport/eglconvenience/qxlibeglintegration.cpp
@@ -41,6 +41,7 @@
 
 #include "qxlibeglintegration_p.h"
 #define QT_DEBUG_X11_VISUAL_SELECTION
+#include <syslog.h>
 VisualID QXlibEglIntegration::getCompatibleVisualId(Display *display, EGLDisplay eglDisplay, EGLConfig config)
 {
     VisualID    visualId = 0;
@@ -96,13 +97,13 @@ VisualID QXlibEglIntegration::getCompati
             if (!visualMatchesConfig) {
                 visualId = 0;
 #ifdef QT_DEBUG_X11_VISUAL_SELECTION
-                qWarning("Warning: EGL suggested using X Visual ID %d (%d %d %d depth %d) for EGL config %d (%d %d %d %d), but this is incompatible",
+                syslog(LOG_WARNING, "Warning: EGL suggested using X Visual ID %d (%d %d %d depth %d) for EGL config %d (%d %d %d %d), but this is incompatible",
                          (int)visualId, visualRedSize, visualGreenSize, visualBlueSize, chosenVisualInfo->depth,
                          configId, configRedSize, configGreenSize, configBlueSize, configAlphaSize);
 #endif
             }
         } else {
-            qWarning("Warning: EGL suggested using X Visual ID %d for EGL config %d, but that isn't a valid ID",
+	    syslog(LOG_WARNING, "Warning: EGL suggested using X Visual ID %d for EGL config %d, but that isn't a valid ID",
                      (int)visualId, configId);
             visualId = 0;
         }
@@ -110,15 +111,15 @@ VisualID QXlibEglIntegration::getCompati
     }
 #ifdef QT_DEBUG_X11_VISUAL_SELECTION
     else
-        qDebug("EGL did not suggest a VisualID (EGL_NATIVE_VISUAL_ID was zero) for EGLConfig %d", configId);
+        syslog(LOG_DEBUG, "EGL did not suggest a VisualID (EGL_NATIVE_VISUAL_ID was zero) for EGLConfig %d", configId);
 #endif
 
     if (visualId) {
 #ifdef QT_DEBUG_X11_VISUAL_SELECTION
         if (configAlphaSize > 0)
-            qDebug("Using ARGB Visual ID %d provided by EGL for config %d", (int)visualId, configId);
+	    syslog(LOG_DEBUG, "Using ARGB Visual ID %d provided by EGL for config %d", (int)visualId, configId);
         else
-            qDebug("Using Opaque Visual ID %d provided by EGL for config %d", (int)visualId, configId);
+	    syslog(LOG_DEBUG, "Using Opaque Visual ID %d provided by EGL for config %d", (int)visualId, configId);
 #endif
         return visualId;
     }
@@ -152,11 +153,11 @@ VisualID QXlibEglIntegration::getCompati
 
     if (visualId) {
 #ifdef QT_DEBUG_X11_VISUAL_SELECTION
-        qDebug("Using Visual ID %d provided by XGetVisualInfo for EGL config %d", (int)visualId, configId);
+        syslog(LOG_DEBUG, "Using Visual ID %d provided by XGetVisualInfo for EGL config %d", (int)visualId, configId);
 #endif
         return visualId;
     }
 
-    qWarning("Unable to find an X11 visual which matches EGL config %d", configId);
+    syslog(LOG_DEBUG, "Unable to find an X11 visual which matches EGL config %d", configId);
     return (VisualID)33;
 }
